### ISSUE TITLE
Unexpected exit

### ISSUE THREAD

**Issue by minyoung90** [UKPLab/sentence-transformers#768]
When I was training a multilingual model (Teacher-Student model as described in document),
I got an unexpected exit after evaluation.

sentence-transformers version == 0.4.1.2
batch size == 32
1 epoch  == 277839 iteration
dev_sentences == 200000
teacher model == custom model
student model == stsb-xlm-r-multilingual
and, using single GPU (T4)

So, when iteration was reached to 199999, started evaluation but even I got evaluation result, program was terminated.
In other words, training was finished after evaluation at first epoch!
What I doubt is there is no log (e.g. out of memory). program was just stopped and didn't save a temporal model at evaluation time.

Now I am trying to train a model with 'dev_sentences == 5000'. if this works, I will report it.

**Comment by nreimers** [comment]
Looks like some error with the GPU which terminated the process. This sometimes happens with Cuda / pytorch

**Comment by minyoung90** [comment]
What I found was that when it was stopped, memory (not gpu) usage almost over 90%. but still had 1~2G enough memory extra space (just my opinion). I just did size up 16G > 32G memory then, I had no unexpected exit. So I am not sure but, If you have problems same like me, I suggest that increase your memory! :) Good luck guys.
